src/main/java/org/entando/plugins/pda/service/ConnectionService.java

```diff
@@ -0,0 +1,145 @@
+package org.entando.plugins.pda.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.entando.plugins.pda.core.engine.Connection;
+import org.entando.plugins.pda.dto.connection.ConfigServiceConnectionsResponse;
+import org.entando.plugins.pda.dto.connection.ConnectionDto;
+import org.entando.plugins.pda.engine.EngineFactory;
+import org.entando.plugins.pda.exception.ConnectionAlreadyExistsException;
+import org.entando.plugins.pda.exception.ConnectionNotFoundException;
+import org.entando.web.exception.HttpException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+public class ConnectionService {
+    private EngineFactory engineFactory;
+    private String configServiceUrl;
+    private RestTemplate restTemplate = new RestTemplate();
+    private Map<String, Connection> connections;
+
+    public ConnectionService(EngineFactory engineFactory, @Value("${config.url}") String configServiceUrl) {
```

Comment:
Any reason not to Autowire the engine factory? Since this is an @Component I don't think it should be getting instantiated directly very often and might be better just to wire them together.

Reply:
This is being `@Autowired`. Spring uses the first constructor to automatically autowire dependencies.

This was changed due to @werbth review comment.